### ISSUE TITLE
How to compile the program into wasm?

### ISSUE THREAD

**Issue by ifsheldon** [sotrh/learn-wgpu#161]
Thank you all for this tutorial! Are you planning to add some basic tutorial for compilation into wasm? Or, should we refer to the Rust WebAssembly book?

**Comment by sotrh** [comment]
WebGPU isn't fully supported for browsers yet so I'm not keen to make a tutorial just yet. The rust WebAssembly book will work just fine, but you'll need to be using Firefox Nightly or Chrome Canary as well as enabling some flags. You can check https://github.com/gpuweb/gpuweb/wiki/Implementation-Status for more details on that. You may need to target a specific branch of wgpu for things to work smoothly though. I'd talk to the guys at https://app.element.io/#/room/#wgpu:matrix.org for more details.

**Comment by ifsheldon** [comment]
Thanks for the information! Please leave this issue opened for anyone who wants to be kept updated.

**Comment by FrankenApps** [comment]
As far as I understand, wgpu-rs >= 0.7 also offers a WebGL2 backend (because this is more or less OpenGL ES 3.0) with WebGL2 beeing support by most browsers by default, except for Safari where it is behind a feature flag. For example there is a [plugin](https://mrk.sed.pl/bevy-showcase/#contributors) for bevy that makes use of this.

**Comment by willmcpherson2** [comment]
Check out out wgpu's wiki page [Running on the Web with WebGPU and WebGL](https://github.com/gfx-rs/wgpu/wiki/Running-on-the-Web-with-WebGPU-and-WebGL).

You can also check out my [repo](https://github.com/willmcpherson2/wu/tree/5415d390bc65b26338de1108082828fb34b83718) which can run on WebGL using this command:

```sh
cargo build --target wasm32-unknown-unknown
wasm-bindgen --out-dir static --target web --no-typescript target/wasm32-unknown-unknown/debug/wu.wasm
```

Then serve the `static` directory, e.g.

```sh
python3 -m http.server -d static
```

**Comment by sotrh** [comment]
Closed by #334
